integration/framework/scenario.go

```diff
@@ -0,0 +1,297 @@
+			"-target":                        "ingester",
+			"-log.level":                     "warn",
+			"-ingester.final-sleep":          "0s",
+			"-ingester.join-after":           "0s",
+			"-ingester.min-ready-duration":   "0s",
+			"-ingester.concurrent-flushes":   "10",
+			"-ingester.max-transfer-retries": "10",
+			"-ingester.num-tokens":           "512",
+			// Configure the ingesters ring backend
+			"-ring.store":      "consul",
+			"-consul.hostname": "consul:8500",
+		}, flags))...),
+		NewReadinessProbe(80, "/ready", 204),
+	))
+}
+
+func (s *Scenario) StartTableManager(name string, flags map[string]string, image string) error {
+	if image == "" {
+		image = getDefaultCortexImage()
+	}
+
+	return s.StartService(NewService(
+		name,
+		image,
+		s.networkName,
+		[]int{80},
+		nil,
+		NewCommandWithoutEntrypoint("cortex", BuildArgs(MergeFlags(map[string]string{
+			"-target":    "table-manager",
+			"-log.level": "warn",
+		}, flags))...),
+		// The table-manager doesn't expose a readiness probe, so we just check if the / returns 404
+		NewReadinessProbe(80, "/", 404),
+	))
+}
+
+func (s *Scenario) StopService(name string) error {
+	service := s.Service(name)
+	if service == nil {
+		return fmt.Errorf("unable to stop service %s because does not exist", name)
+	}
+
+	if err := service.Stop(); err != nil {
+		return err
+	}
+
+	// Remove the service from the list of services
+	for i, entry := range s.services {
+		if entry.name == name {
+			s.services = append(s.services[:i], s.services[i+1:]...)
+			break
+		}
+	}
+
+	return nil
+}
+
+// WaitReady waits until one or more services are ready. A service
+// is ready when its readiness probe succeed. If a service has no
+// readiness probe, it's considered ready without doing any check.
+func (s *Scenario) WaitReady(services ...string) error {
+	for _, name := range services {
+		service := s.Service(name)
+		if service == nil {
+			return fmt.Errorf("Unable to wait for service '%s' ready because the service does not exist", name)
+		}
+
+		if err := service.WaitReady(); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (s *Scenario) Shutdown() {
+	// Kill the services in the opposite order
+	for i := len(s.services) - 1; i >= 0; i-- {
+		if err := s.services[i].Kill(); err != nil {
+			fmt.Println("Unable to kill service", s.services[i].name, ":", err.Error())
+		}
+	}
+
+	// Teardown the docker network
+	if out, err := RunCommandAndGetOutput("docker", "network", "rm", s.networkName); err != nil {
+		fmt.Println(out)
+		fmt.Println("Unable to remove docker network", s.networkName, ":", err.Error())
+	}
+}
+
+// getDefaultCortexImage returns the Docker image to use to run Cortex.
+func getDefaultCortexImage() string {
+	// Get the cortex image from the CORTEX_IMAGE env variable,
+	// falling back to "cortex:latest"
+	if os.Getenv("CORTEX_IMAGE") != "" {
+		return os.Getenv("CORTEX_IMAGE")
+	}
+
+	return "cortex:latest"
```

Comment:
`cortex:latest` doesn't exist on my machine; 
```suggestion
	return "quay.io/cortexproject/cortex:latest"
```